code/game/objects/items/airbubble.dm

```diff
@@ -0,0 +1,462 @@
+// Regular airbubble - folded
+/obj/item/airbubble
+	name = "air bubble"
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+	icon = 'icons/obj/airbubble.dmi'
+	icon_state = "airbubble_fact_folded"
+	w_class = ITEMSIZE_NORMAL
+	var/used = FALSE
+	var/ripped = FALSE
+	var/obj/item/weapon/tank/internal_tank
+
+/obj/item/airbubble/Destroy()
+	qdel(internal_tank)
+	return ..()
+
+/obj/item/airbubble/attack_self(mob/user)
```

Comment:
Put a `if (!isturf(user.loc)) to_chat(user, "You're fucking stupid, the air bubble can't deploy in an enclosed space.")` here.

Otherwise you open it in a locker and then you have a locker in a locker and aaaaaaaaaaaaa.

Reply:
That one is solved